packages/docker/src/plugins/plugin.ts

```diff
@@ -0,0 +1,164 @@
+import {
+  type CreateNodesV2,
+  type ProjectConfiguration,
+  type TargetConfiguration,
+  createNodesFromFiles,
+  readJsonFile,
+  writeJsonFile,
+  CreateNodesContext,
+} from '@nx/devkit';
+import { calculateHashForCreateNodes } from '@nx/devkit/src/utils/calculate-hash-for-create-nodes';
+import { hashObject } from 'nx/src/hasher/file-hasher';
+import { workspaceDataDirectory } from 'nx/src/utils/cache-directory';
+import { existsSync } from 'fs';
+import { dirname, join } from 'path';
+import { getNamedInputs } from '@nx/devkit/src/utils/get-named-inputs';
+
+export interface DockerPluginOptions {
+  buildTarget?: string;
+  runTarget?: string;
+}
+
+interface NormalizedDockerPluginOptions {
+  buildTarget: string;
+  runTarget: string;
+}
+
+type DockerTargets = Pick<ProjectConfiguration, 'targets' | 'metadata'>;
+
+function readTargetsCache(cachePath: string): Record<string, DockerTargets> {
+  return existsSync(cachePath) ? readJsonFile(cachePath) : {};
+}
+
+function writeTargetsCache(
+  cachePath: string,
+  results?: Record<string, DockerTargets>
+) {
+  writeJsonFile(cachePath, results ?? {});
+}
+
+const dockerfileGlob = '**/Dockerfile';
```

Comment:
I think most of our other plugins have something like

`Dockerfile,**/Dockerfile`